components/kyma-environment-broker/internal/process/input/builder.go

```diff
@@ -291,3 +292,58 @@ func mergeMaps(maps ...map[string]struct{}) map[string]struct{} {
 	}
 	return res
 }
+
+func (f *InputBuilderFactory) CreateUpgradeShootInput(pp internal.ProvisioningParameters) (internal.ProvisionerInputCreator, error) {
+	if !f.IsPlanSupport(pp.PlanID) {
+		return nil, errors.Errorf("plan %s in not supported", pp.PlanID)
+	}
+
+	provider, err := f.getHyperscalerProviderForPlanID(pp.PlanID, pp.Parameters.Provider)
```

Comment:
the provider could be changed for a plan (see Trial). Do you expect provider change? what is the expectation of upgradeshoot with  a change of the provider?